File: go/acronym.go
package acronym

import (
	"regexp"
	"strings"
)

func Abbreviate(s string) string {
	re := regexp.MustCompile(`[A-Z]+[a-z']*|[a-z]*`)
	arr := re.FindAll([]byte(s), -1)
	t := []byte{}
	for _, v := range arr {
		if len(v) > 0 {
			t = append(t, v[0])
		}
	}
	return strings.ToUpper(string(t))
}

File: zig/acronym.zig
const std = @import("std");
const ascii = std.ascii;
const mem = std.mem;

pub fn abbreviate(allocator: mem.Allocator, words: []const u8) mem.Allocator.Error![]u8 {
    var arr = std.ArrayList(u8).init(allocator);
    defer arr.deinit();

    var is_in_word = false;
    for (words) |c| {
        if (is_in_word) {
            if (!ascii.isAlphabetic(c) and c != '\'') {
                is_in_word = false;
            }
        } else {
            if (ascii.isAlphabetic(c)) {
                is_in_word = true;
                try arr.append(ascii.toUpper(c));
            }
        }
    }
    return arr.toOwnedSlice();
}
